File: experiment_jp_test/j2pyraw/for__0__DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1.java
class X { static int minPalPartion ( String str ) { int n = str . length ( ) ; int [ ] C = new int [ n ] ; boolean [ ] [ ] P = new boolean [ n ] [ n ] ; int i , j , k , L ; for ( L = 2 ; L <= n ; L ++ ) { for ( i = 0 ; i < n - L + 1 ; i ++ ) { j = i + L - 1 ; if ( L == 2 ) P [ i ] [ j ] = ( str . charAt ( i ) == str . charAt ( j ) ) ; else P [ i ] [ j ] = ( str . charAt ( i ) == str . charAt ( j ) ) && P [ i + 1 ] [ j - 1 ] ; } } for ( i = 0 ; i < n ; i ++ ) { if ( P [ 0 ] [ i ] == true ) C [ i ] = 0 ; else { C [ i ] = Integer . MAX_VALUE ; for ( j = 0 ; j < i ; j ++ ) { if ( P [ j + 1 ] [ i ] == true && 1 + C [ j ] < C [ i ] ) C [ i ] = 1 + C [ j ] ; } } } return C [ n - 1 ] ; }
 }

File: experiment_jp_test/j2pyrawpy/for__0__DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1.py
#!/usr/bin/env python
""" generated source for module for__0__DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1 """
class X(object):
    """ generated source for class X """
    @classmethod
    def minPalPartion(cls, str_):
        """ generated source for method minPalPartion """
        n = len(str_)
        C = [None]*n
        P = [None]*n
        i = int()
        j = int()
        k = int()
        L = int()
        while L <= n:
            while i < n - L + 1:
                j = i + L - 1
                if L == 2:
                    P[i][j] = (str_.charAt(i) == str_.charAt(j))
                else:
                    P[i][j] = (str_.charAt(i) == str_.charAt(j)) and P[i + 1][j - 1]
                i += 1
            L += 1
        while i < n:
            if P[0][i] == True:
                C[i] = 0
            else:
                C[i] = Integer.MAX_VALUE
                while j < i:
                    if P[j + 1][i] == True and 1 + C[j] < C[i]:
                        C[i] = 1 + C[j]
                    j += 1
            i += 1
        return C[n - 1]

